### PR TITLE
Feat/sgid experimental integration

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR introduces new endpoints to support sgid login. To be reviewed in conjunction with PR[#1369](https://github.com/isomerpages/isomercms-frontend/pull/1369) on the isomercms-frontend repo.

This PR introduces a new `SgidAuthRouter`, which introduces 2 new routes. GET `/v2/auth/sgid/auth-url` returns the sgID authentication url, and GET `/api/v3/auth/sgid/auth-redirect` takes two query parameters (`code` and `state`), performs the access token exchange and data retrieval with SGID, and sets the user's session.

A new db table (`sgidLogin`) has also been added to store details of user sessions.

SGID dev docs: https://docs.id.gov.sg/integrations-with-sgid/typescript-javascript

## New Environment Variables
- `SGID_CLIENT_ID`: sgID client ID provided during client registration
- `SGID_CLIENT_SECRET`: sgID client secret provided during client registration
- `SGID_PRIVATE_KEY`: sgID client private key provided during client registration
- `SGID_REDIRECT_URI`: URL to redirect to after authentication with sgID
Note that these new env vars are located in AWS SSM instead of elastic beanstalk, as we would hit the character limit otherwise.

### CODE REVIEW

**Review by lamkeewei — COMMENTED** [review]
Left some comments! Not an expert here, so good to get another opinion!

File: src/routes/v2/sgidAuth.ts
Comment (alexanderleegs) [reply src/routes/v2/sgidAuth.ts]: Thanks for catching this, fixed in https://github.com/isomerpages/isomercms-backend/pull/860/commits/d41bdf62f528a17a84415cdc430b979ff016ed76!

File: src/services/utilServices/SgidAuthService.ts
Comment (alexanderleegs) [reply src/services/utilServices/SgidAuthService.ts:63]: Yup you're right, fixed in https://github.com/isomerpages/isomercms-backend/pull/860/commits/d41bdf62f528a17a84415cdc430b979ff016ed76

File: src/services/utilServices/SgidAuthService.ts
Comment (lamkeewei) [inline src/services/utilServices/SgidAuthService.ts:45]: Technically the state that you pass in the authorization URL should match the `state` that you're storing. Perhaps rather than having the `sessionId` be the `state`, just have it as the `id`?

File: src/services/utilServices/SgidAuthService.ts
Comment (lamkeewei) [inline src/services/utilServices/SgidAuthService.ts:75]: Could be good to verify `state` matches too. This is assuming that you decouple `state` from `sessionId`.

File: src/routes/v2/sgidAuth.ts
Comment (lamkeewei) [inline src/routes/v2/sgidAuth.ts:140]: `loginStatusCode` feels like a weird anti-pattern though. Shouldn't the frontend take the http response status code then redirect accordingly?

File: src/routes/v2/sgidAuth.ts
Comment (lamkeewei) [inline src/routes/v2/sgidAuth.ts:98]: **Optional:** Not sure if it's worthwhile to do an explicit checked for `endsWith('@open.gov.sg')`. Understand that we're relying on the fact that only OGP emails are available for now.

**Comment by kwajiehao** [comment]
As discussed with @alexanderleegs, can consider moving to storing the `codeVerifier` in a cookie (with proper security config) so that you don't need to maintain additional infra. But will leave it to you to decide whether you want to do it in this PR or in subsequent ones

File: src/services/utilServices/SgidAuthService.ts
Comment (alexanderleegs) [reply src/services/utilServices/SgidAuthService.ts]: After chatting with Jie Hao, we've opted to remove the db store completely - https://github.com/isomerpages/isomercms-backend/pull/860/commits/22ce8f90bb6325d20df7d4305f7688f7ff036821

File: src/services/utilServices/SgidAuthService.ts
Comment (alexanderleegs) [reply src/services/utilServices/SgidAuthService.ts]: See above!

**Review by lamkeewei — COMMENTED** [review]
@alexanderleegs Changes lgtm overall! But didn't test this on staging! Will let another Isomer engineer to approve after testing.

**Review by QiluXie — APPROVED** [review]
Tested on staging for several logins and worked well, lgtm! Thanks @alexanderleegs & @harishv7

**Comment by kwajiehao** [comment]
lgtm!

**Pull request merged**
